[Xcode] Symlink to missing headers and check in missing TBDs, instead of mutating the public SDK

emw-apple opened the pull request:
#### d88653e7f78c17f1ce0ad6e2ecb8973762ead39a
<pre>
[Xcode] Symlink to missing headers and check in missing TBDs, instead of mutating the public SDK
<a href="https://bugs.webkit.org/show_bug.cgi?id=259831">https://bugs.webkit.org/show_bug.cgi?id=259831</a>

Reviewed by Alexey Proskuryakov.

Delete the configure-xcode-for-embedded-development logic that copies
headers and TBDs from the public macOS/iOS SDKs. In its place, add a new
script that creates symlinks to needed headers at build time.

When building for an applicable platform against the public SDK, this
script (which runs as part of WTF) creates header symlinks in
$(BUILT_PRODUCTS_DIR)/SDKAdditions, according to a file list provided by
the active additions SDK in WebKitLibraries. That directory is added to
targets SYSTEM_FRAMEWORK_SEARCH_PATHS and SYSTEM_HEADER_SEARCH_PATHS by
settings in the additions SDKs. This avoids needing to configure each
target separately for the SDKAdditions directory, and makes it difficult
to accidentally impact internal builds.

TBDs are not copied via this mechanism, for two reasons:

- They can&apos;t just be symlinked: configure-xcode-for-embedded-development
  was *editing* TBDs as it copied them, to patch their target
  architectures to match all platforms.

- The new linker in Xcode 15 is stricter about objects it links against
  matching the target platform. Even if we copied TBDs at build time and
  mutated their target triples, we&apos;d need a priori knowledge about e.g.
  which frameworks are re-exported by other frameworks on which platforms.

Instead, use the strip-tbd tool added for iOS 17 support to take
existing mutated TBDs, strip them of the symbols we don&apos;t use, and check
them in to the additions SDKs.

For each embedded platform we build for, I ran a command like

    % extract-tbds-from-internal-sdk --ios-device --release

to produce minimal TBDs for iOS 17 simulator in
`WebKitLibraries/SDKs/iphonesimulator17.0-additions.sdk`. This is what
the workflow for keeping these TBDs up to date will look like going
forward. The script&apos;s default values are designed for updating TBDs
based on local build products, but can be customized with arbitrary
build / SDK directories.

* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj: ANGLE needs
  the SDKAdditions directory in place (for Metal), but it does not
  depend on WTF. Add a $(WK_DERIVED_SDK_HEADERS_DIR) input on the
  existing header rule to indicate the dependency. It does nothing in
  internal builds (when the build setting is unset).
* Source/ThirdParty/ANGLE/Configurations/Base.xcconfig: Add back an
  ADDITIONAL_SDK setting that was accidentally removed last time ANGLE
  was synced. (In the future, this will be an EWS failure.)

* Source/WTF/Scripts/SymlinkedHeaders-output.xcfilelist: Added. This is
  an empty file, read by the build system when we&apos;re not building with
  an additions SDK (i.e. internal or macOS).
* Source/WTF/Scripts/SymlinkedHeaders.xcfilelist: Added, ditto above.
* Source/WTF/Scripts/symlink-public-sdk-headers-for-embedded-platforms: Added.
* Source/WTF/WTF.xcodeproj/project.pbxproj: Added build phase.

* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig: Add an
  $(inherited) to SYSTEM_FRAMEWORKS_SEARCH_PATH to prevent paths
  provided by the sparse SDK from being cloberred.
* Tools/MobileMiniBrowser/MobileMiniBrowser.xcodeproj/project.pbxproj:
  For some reason, MobileMiniBrowser now needs to explicitly link
  against UIKit.
* Tools/Scripts/configure-xcode-for-embedded-development: Delete
  header/TBD logic.
* Tools/Scripts/extract-tbds-from-internal-sdk: Rewrite in Python for
  readability and maintainability. Instead of being hard-coded to copy
  private frameworks, this implementation checks against a &quot;base SDK&quot; to
  determine which frameworks need to be copied, so it can find &quot;public&quot;
  frameworks that are mastered out of watchOS/tvOS SDKs.
* Tools/TestWebKitAPI/Configurations/Base.xcconfig: Move $(inherited) to
  the front, as it is in other targets, so that sparse SDK paths are not
  overshadowed by /System/Library/Frameworks.

The rest of the patch is additions SDK noise. In each SDK, I&apos;ve added
default search paths to support finding public framework TBD stubs, as
well as build settings to support the SymlinkedHeaders logic. In some
places (such as the usr/local/include/AvailabilityProhibitedInternal.h
header), I&apos;ve used symlinks from a simulator SDK to the corresponding
device SDK, to cut down on purely duplicate files.

* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/CFNetwork.framework/CFNetwork.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/IOKit.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/Current: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/IOSurface.framework/IOSurface.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/LocalAuthentication.framework/LocalAuthentication.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/Metal.framework/Metal.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/OpenGLES.framework/OpenGLES.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/QuartzCore.framework/QuartzCore.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd:
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk: Removed.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/SDKSettings.plist: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/CFNetwork.framework/CFNetwork.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/IOKit.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/Current: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/IOSurface.framework/IOSurface.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/Metal.framework/Metal.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/OpenGLES.framework/OpenGLES.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/QuartzCore.framework/QuartzCore.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/FileProvider.framework/FileProvider.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/usr: Added.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppServerSupport.framework/AppServerSupport.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppStoreDaemon.framework/AppStoreDaemon.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/ApplePushService.framework/ApplePushService.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AuthKit.framework/AuthKit.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/InstallCoordination.framework/InstallCoordination.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/VisionKitCore.framework/VisionKitCore.tbd:
* WebKitLibraries/SDKs/iphoneos17.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/iphoneos17.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/iphoneos17.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk: Removed.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/SDKSettings.plist: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/CFNetwork.framework/CFNetwork.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/IOKit.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/Current: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/IOSurface.framework/IOSurface.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/Metal.framework/Metal.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/OpenGLES.framework/OpenGLES.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/QuartzCore.framework/QuartzCore.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/AppServerSupport.framework/AppServerSupport.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/AppStoreDaemon.framework/AppStoreDaemon.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/ApplePushService.framework/ApplePushService.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/AuthKit.framework/AuthKit.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/InstallCoordination.framework/InstallCoordination.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/VisionKitCore.framework/VisionKitCore.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/usr: Added.
* WebKitLibraries/SDKs/iphonesimulator17.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/iphonesimulator17.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/iphonesimulator17.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx13.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx13.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx14.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx14.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/CFNetwork.framework/CFNetwork.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/CoreImage.framework/CoreImage.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/IOKit.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/Current: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/IOSurface.framework/IOSurface.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/Metal.framework/Metal.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/OpenGLES.framework/OpenGLES.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/QuartzCore.framework/QuartzCore.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/PepperUICore.framework/PepperUICore.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd:
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk: Removed.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/SDKSettings.plist: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/CFNetwork.framework/CFNetwork.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/CoreImage.framework/CoreImage.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/IOKit.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/Current: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/IOSurface.framework/IOSurface.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/Metal.framework/Metal.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/OpenGLES.framework/OpenGLES.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/QuartzCore.framework/QuartzCore.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/FileProvider.framework/FileProvider.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/PepperUICore.framework/PepperUICore.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/usr: Added.

Canonical link: <a href="https://commits.webkit.org/267073@main">https://commits.webkit.org/267073@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/daf670816291bb607dc864488fb9f607b49f2ac5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15560 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15866 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16229 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17316 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14595 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/15742 "Passed tests") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/18383 "Failed to compile WebKit") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15961 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/17153 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15745 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/23/builds/18383 "Failed to compile WebKit") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13234 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18060 "Built successfully") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/23/builds/18383 "Failed to compile WebKit") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14036 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/20966 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/13355 "Built successfully and passed tests") | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/23/builds/18383 "Failed to compile WebKit") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/14202 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17472 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/14804 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14791 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12532 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/15750 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14047 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3982 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3730 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18409 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/15989 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14611 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/3791 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/WTF/Scripts/symlink-public-sdk-headers-for-embedded-platforms:
```diff
@@ -0,0 +1,61 @@
+    # compilation units.
+    # FIXME: Remove this and the filecmp logic below once iOS/watchOS/tvOS bots
+    # are not using an old, mutated SDK (Fall 2023).
+    sdkroot_dst = os.environ['SDKROOT'] + dst.removeprefix(os.environ['WK_DERIVED_SDK_HEADERS_DIR'])
```
This is a little awkward, how about:
```suggestion
    sdkroot_dst = dst.replace(os.environ['WK_DERIVED_SDK_HEADERS_DIR'], os.environ['SDKROOT'])
```

aproskuryakov commented on Tools/MobileMiniBrowser/Configurations/MobileMiniBrowser.xcconfig:
```diff
@@ -21,6 +21,7 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
 
+OTHER_LDFLAGS = -framework UIKit;
```
Would it be better to add UIKit.framework to the project?

emw-apple replied:
I don't think we have a preferred style, but I don't mind doing this in the UI.

aproskuryakov replied:
I think that having frameworks in the UI reduces confusion for people who have the project open in Xcode.

emw-apple committed:
[Xcode] Symlink to missing headers and check in missing TBDs, instead of mutating the public SDK
https://bugs.webkit.org/show_bug.cgi?id=259831

Reviewed by Alexey Proskuryakov.

Delete the configure-xcode-for-embedded-development logic that copies
headers and TBDs from the public macOS/iOS SDKs. In its place, add a new
script that creates symlinks to needed headers at build time.

When building for an applicable platform against the public SDK, this
script (which runs as part of WTF) creates header symlinks in
$(BUILT_PRODUCTS_DIR)/SDKAdditions, according to a file list provided by
the active additions SDK in WebKitLibraries. That directory is added to
targets SYSTEM_FRAMEWORK_SEARCH_PATHS and SYSTEM_HEADER_SEARCH_PATHS by
settings in the additions SDKs. This avoids needing to configure each
target separately for the SDKAdditions directory, and makes it difficult
to accidentally impact internal builds.

TBDs are not copied via this mechanism, for two reasons:

- They can't just be symlinked: configure-xcode-for-embedded-development
  was *editing* TBDs as it copied them, to patch their target
  architectures to match all platforms.

- The new linker in Xcode 15 is stricter about objects it links against
  matching the target platform. Even if we copied TBDs at build time and
  mutated their target triples, we'd need a priori knowledge about e.g.
  which frameworks are re-exported by other frameworks on which platforms.

Instead, use the strip-tbd tool added for iOS 17 support to take
existing mutated TBDs, strip them of the symbols we don't use, and check
them in to the additions SDKs.

For each embedded platform we build for, I ran a command like

    % extract-tbds-from-internal-sdk --ios-device --release

to produce minimal TBDs for iOS 17 simulator in
`WebKitLibraries/SDKs/iphonesimulator17.0-additions.sdk`. This is what
the workflow for keeping these TBDs up to date will look like going
forward. The script's default values are designed for updating TBDs
based on local build products, but can be customized with arbitrary
build / SDK directories.

* Source/ThirdParty/ANGLE/ANGLE.xcodeproj/project.pbxproj: ANGLE needs
  the SDKAdditions directory in place (for Metal), but it does not
  depend on WTF. Add a $(WK_DERIVED_SDK_HEADERS_DIR) input on the
  existing header rule to indicate the dependency. It does nothing in
  internal builds (when the build setting is unset).
* Source/ThirdParty/ANGLE/Configurations/Base.xcconfig: Add back an
  ADDITIONAL_SDK setting that was accidentally removed last time ANGLE
  was synced. (In the future, this will be an EWS failure.)

* Source/WTF/Scripts/SymlinkedHeaders-output.xcfilelist: Added. This is
  an empty file, read by the build system when we're not building with
  an additions SDK (i.e. internal or macOS).
* Source/WTF/Scripts/SymlinkedHeaders.xcfilelist: Added, ditto above.
* Source/WTF/Scripts/symlink-public-sdk-headers-for-embedded-platforms: Added.
* Source/WTF/WTF.xcodeproj/project.pbxproj: Added build phase.

* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig: Add an
  $(inherited) to SYSTEM_FRAMEWORKS_SEARCH_PATH to prevent paths
  provided by the sparse SDK from being cloberred.
* Tools/MobileMiniBrowser/MobileMiniBrowser.xcodeproj/project.pbxproj:
  For some reason, MobileMiniBrowser now needs to explicitly link
  against UIKit.
* Tools/Scripts/configure-xcode-for-embedded-development: Delete
  header/TBD logic.
* Tools/Scripts/extract-tbds-from-internal-sdk: Rewrite in Python for
  readability and maintainability. Instead of being hard-coded to copy
  private frameworks, this implementation checks against a "base SDK" to
  determine which frameworks need to be copied, so it can find "public"
  frameworks that are mastered out of watchOS/tvOS SDKs.
* Tools/TestWebKitAPI/Configurations/Base.xcconfig: Move $(inherited) to
  the front, as it is in other targets, so that sparse SDK paths are not
  overshadowed by /System/Library/Frameworks.

The rest of the patch is additions SDK noise. In each SDK, I've added
default search paths to support finding public framework TBD stubs, as
well as build settings to support the SymlinkedHeaders logic. In some
places (such as the usr/local/include/AvailabilityProhibitedInternal.h
header), I've used symlinks from a simulator SDK to the corresponding
device SDK, to cut down on purely duplicate files.

* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/CFNetwork.framework/CFNetwork.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/IOKit.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/Current: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/IOSurface.framework/IOSurface.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/LocalAuthentication.framework/LocalAuthentication.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/Metal.framework/Metal.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/OpenGLES.framework/OpenGLES.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/Frameworks/QuartzCore.framework/QuartzCore.tbd: Added.
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd:
* WebKitLibraries/SDKs/appletvos16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd:
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk: Removed.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/SDKSettings.plist: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/CFNetwork.framework/CFNetwork.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/IOKit.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/Current: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/IOSurface.framework/IOSurface.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/Metal.framework/Metal.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/OpenGLES.framework/OpenGLES.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/Frameworks/QuartzCore.framework/QuartzCore.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/FileProvider.framework/FileProvider.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Added.
* WebKitLibraries/SDKs/appletvsimulator16.0-additions.sdk/usr: Added.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppServerSupport.framework/AppServerSupport.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppStoreDaemon.framework/AppStoreDaemon.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/ApplePushService.framework/ApplePushService.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/AuthKit.framework/AuthKit.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/InstallCoordination.framework/InstallCoordination.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd:
* WebKitLibraries/SDKs/iphoneos16.0-additions.sdk/System/Library/PrivateFrameworks/VisionKitCore.framework/VisionKitCore.tbd:
* WebKitLibraries/SDKs/iphoneos17.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/iphoneos17.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/iphoneos17.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk: Removed.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/SDKSettings.plist: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/CFNetwork.framework/CFNetwork.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/IOKit.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/Current: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/IOSurface.framework/IOSurface.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/Metal.framework/Metal.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/OpenGLES.framework/OpenGLES.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/Frameworks/QuartzCore.framework/QuartzCore.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/AppServerSupport.framework/AppServerSupport.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/AppStoreDaemon.framework/AppStoreDaemon.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/ApplePushService.framework/ApplePushService.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/AuthKit.framework/AuthKit.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/FrontBoardServices.framework/FrontBoardServices.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/InstallCoordination.framework/InstallCoordination.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/System/Library/PrivateFrameworks/VisionKitCore.framework/VisionKitCore.tbd: Added.
* WebKitLibraries/SDKs/iphonesimulator16.0-additions.sdk/usr: Added.
* WebKitLibraries/SDKs/iphonesimulator17.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/iphonesimulator17.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/iphonesimulator17.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx13.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx13.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx14.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/macosx14.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/CFNetwork.framework/CFNetwork.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/CoreImage.framework/CoreImage.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/IOKit.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/Current: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/IOSurface.framework/IOSurface.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/Metal.framework/Metal.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/OpenGLES.framework/OpenGLES.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/Frameworks/QuartzCore.framework/QuartzCore.tbd: Added.
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/IOSurfaceAccelerator.framework/IOSurfaceAccelerator.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/PepperUICore.framework/PepperUICore.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd:
* WebKitLibraries/SDKs/watchos9.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd:
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk: Removed.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/SDKSettings.plist: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/SymlinkedHeaders-output.xcfilelist: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/SymlinkedHeaders.xcfilelist: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/CFNetwork.framework/CFNetwork.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/CoreImage.framework/CoreImage.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/IOKit.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/IOKit.framework/Versions/Current: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/IOSurface.framework/IOSurface.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/Metal.framework/Metal.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/OpenGLES.framework/OpenGLES.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/Frameworks/QuartzCore.framework/QuartzCore.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/AppSupport.framework/AppSupport.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/CorePrediction.framework/CorePrediction.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/FileProvider.framework/FileProvider.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/FontServices.framework/libFontParser.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/PepperUICore.framework/PepperUICore.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/PrototypeTools.framework/PrototypeTools.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/RunningBoardServices.framework/RunningBoardServices.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/SafariSafeBrowsing.framework/SafariSafeBrowsing.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/System/Library/PrivateFrameworks/URLFormatting.framework/URLFormatting.tbd: Added.
* WebKitLibraries/SDKs/watchsimulator9.0-additions.sdk/usr: Added.

Canonical link: https://commits.webkit.org/267073@main
